scripts/process/process-hydrograph.R

```diff
@@ -61,39 +60,46 @@ process.plot_hydrographTotal <- function(viz=getContentInfo("NMHydrograhTotal-sv
     rects <- xml_find_all(rect_svg_all, '//*[local-name()="rect"]')
     rect_svg <- rects[length(rects)]
     
-    xml_set_attr(rect_svg, 'id', paste0("b",yr))
+    xml_set_attr(rect_svg, 'data-year', as.character(yr))
     xml_attr(rect_svg, "style") <- NULL
     xml_attr(rect_svg, "clip-path") <- NULL
-    xml_attr(rect_svg, "clip-path") <- NULL
     xml_attr(rect_svg, "class") <- "years-rect"
-    xml_attr(rect_svg, "class") <- "years-rect"
-    xml_attr(rect_svg, "onmouseover") <- paste0("vizlab.showline('",yr,"')")
-    xml_attr(rect_svg, "onmouseout") <- paste0("vizlab.greyline('",yr,"')")
 
     xml_add_child(g.year_rects, rect_svg[[1]])
   }
   
   g.totalPoly <- xml_add_sibling(xml_children(total_svg)[[length(xml_children(total_svg))]], 
                                  'g', id='totalHydro','class'='total-hydrograph')
   
-  hydro_lines <- svglite::xmlSVG({
-
-    par(omi=c(0,0,0,0), mai=c(0.5,0.75,0,0),las=1, xaxs = "i")
-    plot(rawData, 
-         type="l", axes=F, ann=F, xaxt="n")
-  }, height=height, width=width)
+  full_range <- as.Date(c(paste0(yearRange[1],"-01-01"),
+                          paste0(yearRange[2],"-12-31")))
   
-  pline <- xml_find_first(hydro_lines, '//*[local-name()="polyline"]')
-  xml_remove(pline)
-  xml_attr(pline,"class") <- "total-hydrograph"
-  xml_attr(pline,"clip-path") <- NULL
-  xml_attr(pline,"style") <- NULL
+  for(i in 1:length(date_chunks)){
+    x <- date_chunks[[i]]
+    y <- flow_chunks[[i]]
+    
+    chunk_data <- data.frame(approx(x, y, 
+                                   n = length(x)/viz[["full-desample"]]))
+    chunk_data$x <- as.Date(chunk_data$x, origin="1970-01-01")
+    
+    hydro_lines <- svglite::xmlSVG({
   
-  xml_add_child(g.totalPoly, pline)
+      par(omi=c(0,0,0,0), mai=c(0.5,0.75,0,0),las=1, xaxs = "i")
+      plot(chunk_data, xlim=full_range,ylim=c(0, yMax),
+           type="l", axes=F, ann=F, xaxt="n")
+    }, height=height, width=width)
+    
+    pline <- xml_find_first(hydro_lines, '//*[local-name()="polyline"]')
+    xml_remove(pline)
+    xml_attr(pline,"class") <- "total-hydrograph"
+    xml_attr(pline,"clip-path") <- NULL
+    xml_attr(pline,"style") <- NULL
+    
+    xml_add_child(g.totalPoly, pline)
```

Comment:
This may be a pain here to do, but it is nice to add the class `svg-text` to any of the text we have in the svg. This keeps it from being "selectable", which can get pretty funky with the interactions.